Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -859,21 +859,22 @@ class BrowserViewController: UIViewController {
     override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey: Any]?, context: UnsafeMutableRawPointer?) {
         let webView = object as! WKWebView
         guard let path = keyPath else { assertionFailure("Unhandled KVO key: \(keyPath ?? "nil")"); return }
+
+        guard tabManager.selectedTab?.webView?.matches(webView) ?? false else {
```

Comment:
I might be inclined to wrap this `matches` again, twice: once on `selectedTab`:

```
selectedTab?.usesWebView(webView)
```

and then again on `TabManager`:

```
guard tabManager.isSelectedWebView(webView)
```